src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -388,7 +372,7 @@ class QualifyClauseSegment(BaseSegment):
 
 
 @snowflake_dialect.segment(replace=True)
-class SelectStatementSegment(ansi_SelectClauseSegment):
+class SelectStatementSegment(ansi.SelectStatementSegment):
```

Comment:
Why did the base class change from `SelectClauseSegment` to `SelectStatementSegment`?

Reply:
it was the wrong class in the first place, but because it wasn't registered properly nobody had noticed 🤷